src/lib/spinel/radio_spinel.cpp

```diff
@@ -2113,6 +2113,8 @@ void RadioSpinel::RecoverFromRcpFailure(void)
     else
     {
         ResetRcp(mResetRadioOnStartup);
+	// this clear is necessary in case the RCP has sent messages between disable and reset
+	mRxFrameBuffer.Clear();
```

Comment:
If we clear again here, how can we be sure we are not clearing spinel frames emitted by RCP after reset?

Reply:
From RCP side, We are clearing NCP buffer on reset command. I will raise separate PR for it or shall I do along with this PR only?

Reply:
> From RCP side, We are clearing NCP buffer on reset command. 

Yes, adding this may be helpful. But I expect on most platform RCP/NCP (and its tx queue) to be cleared/reset from reset command itself?

----

My question is about spinel messages emitted after "reset" command processed by RCP. Basically if we rx spinel msg after and in between the `ResetRcp(mResetRadioOnStartup)` call and the next line where `mRxFrameBuffer.Clear()` is now done? If I am not mistaken , the spinel frame rx can be async and can happen in parallel to `RecoverFromRcpFailure()` execution?

Reply:
we can dropped all spinel frame when we received reset command on RCP. I have added those changes in PR.

Reply:
@abtink, can you please review latest change?

Reply:
Thanks. But I think my question may be still valid. It is about messages sent after "reset" from RCP (e.g. there may be logs for crash dumpt or other events emitted by RCP that are immediately sent after RESET event).

If we do `mRxFrameBuffer.Clear()` here after `ResetRcp(mResetRadioOnStartup)` we may unintentionally drop them?

Reply:
One idea/suggestion:
- How about adding `mRxFrameBuffer.Clear()` in `RadioSpinel::HandleValueIs()` where we process the received "RESET" event, i.e here https://github.com/openthread/openthread/blob/ffe2f5257989b2e72fbb0ac96e06098312a2671f/src/lib/spinel/radio_spinel.cpp#L665-L675

Reply:
I am talking about frames sent by RCP **after** it sends the "RESET". We should not drop those.

Reply:
@abtink,

Yes, It is fine to clear when reset event received. I have made changes and updated PR. Please review it.

Reply:
@abtink ,
can you please review changes?